pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/BaseMailboxReceiveOperator.java

```diff
@@ -59,13 +59,15 @@ public BaseMailboxReceiveOperator(OpChainExecutionContext context, RelDistributi
     int workerId = context.getServer().workerId();
     MailboxMetadata senderMailBoxMetadatas =
         context.getStageMetadata().getWorkerMetadataList().get(workerId).getMailBoxInfosMap().get(senderStageId);
-    Preconditions.checkState(senderMailBoxMetadatas != null && !senderMailBoxMetadatas.getMailBoxIdList().isEmpty(),
-        "Failed to find mailbox for stage: %s",
-        senderStageId);
-    _mailboxIds = MailboxIdUtils.toMailboxIds(requestId, senderMailBoxMetadatas);
-    _mailboxes = _mailboxIds.stream()
-        .map(mailboxId -> _mailboxService.getReceivingMailbox(mailboxId))
-        .collect(Collectors.toCollection(ArrayDeque::new));
+    if (senderMailBoxMetadatas != null && !senderMailBoxMetadatas.getMailBoxIdList().isEmpty()) {
```

Comment:
Preconditions.checkState((senderMailBoxMetadatas != null && !senderMailBoxMetadatas.getMailBoxIdList().isEmpty()) || expectEmptySendingMailbox) ...

and set the expectEmptySendingMailbox from MailboxAssignmentVisitor when previous stage has no server assigned. ?

Reply:
discussed offline. this should be allowed. instead we should make a empty server check on WorkerManager to ensure that no empty server plan for intermediate stage is allowed

Reply:
Need to modify some tests for this behavior.